File: src/Messages/Light/StateInfrared.java
package Messages.Light;

import LifxCommander.CommonMethods;
import Messages.DataTypes.Payload;

public class StateInfrared extends Payload {
	int code = 121;
	int brightness;			// 16-Bits (Unsigned)
	
	public StateInfrared() {
		brightness = 0;
	}
	
	public StateInfrared(int brightness) {
		this.brightness = brightness;
	}
	
	public StateInfrared(StateInfrared stateInfrared) {
		brightness = stateInfrared.brightness;
	}
	
	public int getCode() {
		return code;
	}
	
	public int getBrightness() {
		return brightness;
	}
	
	public void setBrightness(int brightness) {
		this.brightness = brightness;
	}
	
	public void setFromCommandByteArray(byte[] byteArray) {
		String brightnessBinStr = CommonMethods.convertByteToBinaryString(byteArray[36]);
		brightness = Integer.parseInt(brightnessBinStr, 2);
	}
}

File: src/main/java/org/stevenlowes/tools/lifxcontroller/commands/response/light/StateInfrared.kt
package org.stevenlowes.tools.lifxcontroller.commands.response.light

import org.stevenlowes.tools.lifxcontroller.Utils
import org.stevenlowes.tools.lifxcontroller.commands.response.ResponsePayload

class StateInfrared(var brightness: Int = 0) : ResponsePayload(121) {
    override fun setFromCommandByteArray(byteArray: ByteArray) {
        val brightnessBinStr = Utils.convertByteToBinaryString(byteArray[36])
        brightness = Integer.parseInt(brightnessBinStr, 2)
    }
}
